sources/osgShader/Compiler.js

```diff
@@ -697,7 +710,7 @@ define( [
 
             }
             // TODO: shadow Attributes in node, is this the legit way
```

Comment:
That's the point it's a big subject to discuss, not easy to keep the this part flexible, because it almost needs to know all externals nodeType or at least big sub system to handle the logic here.
Problably we would need to debrief about this point, that's a very important one.